ERCS/erc-7836.md

```diff
@@ -0,0 +1,188 @@
+    value: `0x${string}`
+  }[];
+  capabilities: Record<string, any>
+}]
+```
+
+##### Example Parameters
+
+```typescript!
+wallet_prepareCalls([{
+  version: '1.0',
+  from: '0x...',
+  chainId: '0x...',
+  calls: [{
+    to: '0x...'
+    data: '0x...'
+    value: '0x...'
+  }],
+  capabilities: {
+    paymasterService: {
+      url: 'https://...'
+    }
+  }
+}])
+```
+
+#### Return Value
+
+```typescript!
+type PrepareCallsReturnValue = {
+  type: string
+  chainId: `0x${string}`
+  signatureRequest: {
+    hash: `0x${string}`
+  }
+  capabilities: Record<string, any>
+}
+##### Example Return Value
+
+```typescript!
+{
+  type: 'user-operation-v06',
+  userOp: {
+    sender: '0x...',
+    ...
+  },
+  chainId: '0x01',
+  signatureRequest: {
+    hash: '0x...'
+  }
+  capabilities: {
+    paymasterService: {
+      sponsor: 'My app'
+    }
+  }
+}
+After calling `wallet_prepareCalls`, app developers are expected to sign the hash provided in the `signatureRequest` field and submit the prepared and signed calls back to the wallet (either via its [EIP-1193](./eip-1193.md) provider or to its server counterpart) using the second RPC method, `wallet_sendPreparedCalls`.
+
+### `wallet_sendPreparedCalls`
+
+Accepts prepared calls from the response to a `wallet_prepareCalls` request along with a signature, and returns an EIP-5792 call bundle ID.
+
+#### Parameters
+
+```typescript!
+type SendPreparedCallsParams = [{
+  version: string
+  type: string
+  data: any
+  chainId: `0x${string}`
+  signature: {
+    type: string
+    signature: `0x${string}`
+  }
+}]
+```
+
+##### Example Parameters
+
+```typescript!
+wallet_sendPreparedCalls([{
+  version: '1.0',
+  type: 'user-operation-v06',
+  data: {
+    sender: '0x...',
+    ...
+  },
+  chainId: '0x01',
+  signature: {
+    type: 'secp256k1',
+    data: '0x...'
+  }
+}])
+```
+
+#### Return Value
+
+```typescript!
+type SendPreparedCallsReturnValue = {
+  preparedCallIds: string[]
+}
+```
+
+##### Example Return Value
+
+```typescript!
+['0x...']
+```
+
+The `wallet_sendPreparedCalls` RPC responds with EIP-5792 call bundle identifiers, so apps that already use `wallet_sendCalls` to submit calls to the wallet can continue to use `wallet_getCallsStatus` to get the status of submitted calls.
+
+## Rationale
+
+Many session key use cases involve submitting operations on users' behalves. One example is an application that automates a user's DEX trading. In these cases, the wallet may provide a "wallet server" URL for the app to submit JSON-RPC requests to outside of the context of a browser.
+
+It is not in the scope of this EIP, but someone could define a `walletServer` [EIP-5792](./eip-5792.md) capability that declares a public wallet server URL.
+
+```json
+{
+  "0x2105": {
+    "walletServer": {
+      "supported": true,
+      "url": "https://.../0x2105"
+    }
+  },
+  "0x14A34": {
+    "walletServer": {
+      "supported": true,
+      "url": "https://.../0x14A34"
+    }
+  }
+}
```

Comment:
Suggest referencing standardized [CAIP-345](https://github.com/ChainAgnostic/CAIPs/pull/345) as the example. We would appreciate feedback on CAIP-345 and how it interops with ERC-7836. This is what we are using for WalletConnect connections.

```suggestion
{
  "0x2105": {
    "walletService": {
      "url": "https://.../0x2105",
      "methods": ["wallet_getCallsStatus"]
    }
  },
  "0x14A34": {
    "walletService": {
      "url": "https://.../0x14A34",
      "methods": ["wallet_getCallsStatus"]
    }
  }
}
```